src/names.md

```diff
@@ -0,0 +1,133 @@
+# Names
+
+Entities declare a *name* to refer to that entity. Some entities are
+[explicitly declared](#explicit-entities) in the source code, and some are
+[implicitly declared](#implicit-entities) as part of the language or compiler
+extensions.
+
+Entity names are valid within a [*scope*] — a region of source text where that
+name may be referenced.
+
+[*Paths*] are used to refer to an entity, possibly in another scope. Lifetimes
+and loop labels use a [dedicated syntax][lifetimes-and-loop-labels] using a
+leading quote.
+
+Names are segregated into different [*namespaces*], allowing entities in
+different namespaces to share the same name without conflict.
+
+[*Name resolution*] is the compile-time process of tying paths, identifiers,
+and labels to entity declarations.
+
+Access to certain names may be restricted based on their [*visibility*].
+
+## Explicit entities
+
+Entities that explicitly introduce a name in the source code are:
+
+* [Items]
```

Comment:
Yea, I think I had that originally, but removed it due to the risk that markdown would "eat" it.  For example, if you have a single word after the colon, like `* [Boolean type]: `bool``, it just disappears since it is interpreted as a link reference. It should be safe otherwise, though.